uiowa/uiowa
# 2 plus 2 continued

Resolves #5157  

As a prospective student interested in a 2 plus 2 plan, I want to see a list of participating colleges for a given major, so that I can more easily explore my options.

As a prospective student learning more about my area of study, I want to see all of my options (including transfer options) if they are available, so that I can compare all my options.

As an admissions site owner, I want the relevant connections between 2 plus 2 plans, areas of study, and majors to be built automatically, so that after creating my content the structure and connections are already in place with little room for user error.





# How to test



- `git checkout 2-plus-2-continued && git pull`
- `blt ds --site=admissions.uiowa.edu`
- Create some Community College taxonomy terms
- Create some Major (2 plus 2) content pieces connecting community colleges to areas of study
- Check the AoS page, and if there is a Transfer Tips card present, see that it is linked to 2 plus 2 view
  - The view should have a listing of all (published) community colleges which offer majors related to this area of study
  - Each listed college link should direct to the specific major page related to that AoS and community college
- On a transfer tips page, if there is a community college which offers a major related to the area of study, there should be a link to the above view page
- Check that various page titles, links, url paths, appear relevant across throughout the above connections
- Check that the following AoSs are usable by creating associated majors and checking their corresponding views pages
   - https://admissions.uiowa.ddev.site/academics/nursing-rn-bsn
   - https://admissions.uiowa.ddev.site/academics/resilience-and-trauma-informed-perspectives-certificate
   - https://admissions.uiowa.ddev.site/academics/interscholastic-athleticactivities-administration

## Note

~There is a slight inconsistency between pages of some areas of study (possible in 25/150 of them that have "of" or "and" in their title). In some of these cases, the URL path is sometimes constructed as "/asian-languages-literature/" and sometimes "/asian-languages-and-literature/". We may want to create a followup task for this specifically, but it touches content already in production such as Transfer Tips, and needs more exploration.~

Test links across the content structure for functionality, but some of the inconsistencies may be a result of paths being created both by pathauto and user-generated. https://github.com/uiowa/uiowa/pull/5199#issuecomment-1118684449

## Follow-up

Breadcrumb to be updated after a landing page is created. To come later.
